vllm/v1/worker/gpu_model_runner.py

```diff
@@ -2054,6 +2068,13 @@ def get_kv_cache_spec(self) -> dict[str, KVCacheSpec]:
         use_mla = self.vllm_config.model_config.use_mla
         kv_cache_spec: dict[str, KVCacheSpec] = {}
         for layer_name, attn_module in layers.items():
+            if (kv_tgt_layer :=
+                    attn_module.kv_sharing_target_layer_name) is not None:
+                validate_kv_target_layer(layer_name, kv_tgt_layer, layers)
+                # KV cache is shared with earlier layer, don't create a spec
```

Comment:
1. Can you explain how kv sharing is implemented here?
2. Can you merge the validation here to those in `attention/layer.py`? If the code for validation is too long, I suggest to create a utility function there.

Reply:
You can access the target layer with this static forward context. Is it enought?
https://github.com/vllm-project/vllm/blob/1661a9c28f090505fe127ca4cb82678486a523cc/vllm/attention/layer.py#L152